front-end/src/components/main-page/TranscriptUploadModal.jsx

```diff
@@ -47,10 +47,23 @@ class TranscriptUploadModal extends React.Component {
     }
   }
 
+  checkProperties(intent) {
+    return (!intent.project_id || !intent.trace_type || !intent.trace_payload);
+  }
+
   validateData(data) {
     try {
       const parsedData = JSON.parse(data);
-      return parsedData.data;
+      if (parsedData.data){
+        for (const intent of parsedData.data) {
+          if (this.checkProperties(intent)){
+            return false;
+          }
+        }
+        return true;
+      } else {
+        return false;
+      }
```

Comment:
I forgot to click confirm on my nitpick (sorry!)

So basically I think checkProperties could potentially be a bit better scoped/named. See below:

```
alidateData(data) {
const intentIsMissingFields = (intent) => {
    return (!intent.project_id || !intent.trace_type || !intent.trace_payload)
}
    try {
      const parsedData = JSON.parse(data);
      return parsedData.data;
      if (parsedData.data){
        for (const intent of parsedData.data) {
          if (intentIsMissingFields (intent)){
            return false;
          }
        }
        return true;
      } else {
        return false;
      }
     } catch (error) {
      return false;
    }
```

Bonus: we could define this new arrow function at the top of the file, export it, and test it (which might be nice, if a little extra)

Reply:
Let me know if anything here is unclear